Source/SteveSloth/Private/MyPossumEnemyClass.cpp

```diff
@@ -0,0 +1,114 @@
+/****************************************************************************************
+ * Copyright: SteveSloth
+ * Name: Elad Saretzky
+ * Script: MyPossumEnemyClass.cpp
+ * Date: May 6, 2024
+ * Description: Does all of the specific to possum things
+ * TODO:
+ * Known Bugs:
+ ****************************************************************************************/
+
+
+#include "MyPossumEnemyClass.h"
+
+AMyPossumEnemyClass::AMyPossumEnemyClass()
+{
+	IsIdle = false;
+	IsChasing = false;
+	IsAttackingMelee = false;
+	IsAttackingRanged = false;
+}
+
+void AMyPossumEnemyClass::BeginPlay()
+{
+	Super::BeginPlay();
+}
+
+void AMyPossumEnemyClass::Tick(float DeltaTime)
+{
+	Super::Tick(DeltaTime);
+
+	//Linetrace:
+	FHitResult outHit;
+
+	FVector start = this->GetActorLocation(); //this should change to a specific location we want to shoot from (arm maybe)
+	FVector end = (start + (this->GetActorForwardVector() * RangedAttackRange));
+
+	FCollisionQueryParams collisionParams;
+
+	bool isHit = GetWorld()->LineTraceSingleByChannel(outHit, start, end, ECC_Visibility, collisionParams);
```

Comment:
here add
collisionParams.AddIgnoredActor(this);
to avoid self collision checks